c_src/procket.c

```diff
@@ -134,6 +136,24 @@ nif_fdrecv(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
             enif_make_int(env, s));
 }
 
+static ERL_NIF_TERM
+nif_setns(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
+{
+    int fd;
+    ErlNifBinary buf = {0};
+
+    if (!enif_inspect_binary(env, argv[0], &buf))
+        return enif_make_badarg(env);
+
+    fd = open((const char*)buf.data, O_RDONLY);  /* Get descriptor for namespace */
+    if (fd < 0)
+        return error_tuple(env, errno);
+
+    if (setns(fd, 0) == -1)
+        return error_tuple(env, errno);
+
```

Comment:
`fd` can be closed after setting the namespace. If setns() fails, it should be closed before returning the error.